### PR TITLE
[MU4] Notation parts

### PR DESCRIPTION
**Pull request by Eism**

### CODE REVIEW

File: libmscore/part.cpp [hunk 1]
```diff
@@ -34,6 +36,7 @@ namespace Ms {
 Part::Part(Score* s)
     : ScoreElement(s)
 {
+    _id = QUuid::createUuid().toString();
```
Comment (igorkorsukov) [inline libmscore/part.cpp:39]: Maybe better use just a static variable?, it is better for debugging and performance

**Review by RobFog — COMMENTED** [review]
"instuments" should read "instruments".

**Comment by Eism** [comment]
> "instuments" should read "instruments".

@RobFog 
Please explain what you mean?

**Comment by RobFog** [comment]
> > "instuments" should read "instruments".
> 
> @RobFog
> Please explain what you mean?

The title of your last commit has a typo.

**Comment by RobFog** [comment]
> > > "instuments" should read "instruments".
> > 
> > 
> > @RobFog
> > Please explain what you mean?
> 
> The title of your last commit has a typo.

It's still there.

File: mu4/domain/notation/internal/notationparts.h [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    async::Channel<const Part*> partChanged() const override;
+    async::Channel<const Instrument*> instrumentChanged() const override;
+    async::Channel<const Staff*> staffChanged() const override;
+    async::Notification partsChanged() const override;
+
+private:
+    struct InstrumentInfo
+    {
+        int tick;
+        Instrument* instrument;
```
Comment (RomanPudashkin) [inline mu4/domain/notation/internal/notationparts.h:73]: Need default values

File: libmscore/part.cpp (hunk 1 above)
Comment (RomanPudashkin) [reply libmscore/part.cpp]: fixed

**Comment by Jojo-Schmitz** [comment]
simple code style fix needed

**Comment by RobFog** [comment]
Rebase needed.

**Comment by RobFog** [comment]
> > > > "instuments" should read "instruments".
> > > 
> > > 
> > > @RobFog
> > > Please explain what you mean?
> > 
> > 
> > The title of your last commit has a typo.
> 
> It's still there.
> 
> Annotated screenshot:
> Bildschirmfoto 2020-08-25 um 11 49 39

Still there.

**Comment by Jojo-Schmitz** [comment]
2 simple coding style issue prevent the CI_MacOS MU4 build to succeed

File: libmscore/part.cpp [hunk 3]
```diff
@@ -333,6 +336,15 @@ void Part::removeInstrument(const Fraction& tick)
     _instruments.erase(i);
 }
 
+void Part::removeInstrument(const QString& instrumentId)
+{
+    for (auto it = _instruments.begin(); it != _instruments.end(); it++) {
+        if (it->second->instrumentId() == instrumentId) {
+            _instruments.erase(it);
```
Comment (igorkorsukov) [inline libmscore/part.cpp:343]: better add break

**Review by igorkorsukov — APPROVED** [review]

File: libmscore/part.cpp (hunk 3 above)
Comment (RomanPudashkin) [reply libmscore/part.cpp:343]: fixed

**Comment by Eism** [comment]
> > > > "instuments" should read "instruments".
> > > 
> > > 
> > > @RobFog
> > > Please explain what you mean?
> > 
> > 
> > The title of your last commit has a typo.
> 
> It's still there.
> 
> Annotated screenshot:
> Bildschirmfoto 2020-08-25 um 11 49 39

fixed

**Pull request merged**